server/api/src/main/java/org/hawkular/apm/server/api/task/ProcessingUnit.java

```diff
@@ -102,32 +104,40 @@ public void handle(String tenantId, List<T> items) throws Exception {
 
         processor.initialise(tenantId, items);
 
-        for (int i = 0; i < items.size(); i++) {
+        if (processor.getType() == ProcessorType.ManyToMany) {
             try {
-                if (processor.isMultiple()) {
-                    List<R> result = processor.processMultiple(tenantId, items.get(i));
-                    if (resultHandler != null && result != null && !result.isEmpty()) {
-                        if (results == null) {
-                            results = new ArrayList<R>();
+                results = processor.processManyToMany(tenantId, items);
+            } catch (Exception e) {
+                retries = items;
+            }
+        } else {
+            for (int i = 0; i < items.size(); i++) {
```

Comment:
Although I haven't tried it to be sure, my concern would be performance, e.g. https://jaxenter.com/java-performance-tutorial-how-fast-are-the-java-8-streams-118830.html

I can create a separate jira to investigate this?